### PR TITLE
feat(helm): add in ability to inject init containers in to deployment from values

### PR DESCRIPTION
**Pull request by chrismellard**

### CODE REVIEW

File: charts/kubernetes-external-secrets/values.yaml [hunk 1]
```diff
@@ -123,3 +123,9 @@ serviceMonitor:
   enabled: false
   interval: "30s"
   namespace:
+
+deploymentInitContainers: {}
+
+emptyDirVolumes: {}
```
Comment (Flydiverny) [inline charts/kubernetes-external-secrets/values.yaml:129]: Can we do more generic and add `extraVolumeMounts` and `extraVolumes` instead ? :)

Comment (chrismellard) [reply charts/kubernetes-external-secrets/values.yaml]: Updated the PR to reflect your suggestion. Thanks for the review 👍

File: charts/kubernetes-external-secrets/values.yaml [hunk 2]
```diff
@@ -125,3 +125,19 @@ serviceMonitor:
...
+# Add in additional named volumes and volume mounts to the deployment
+#
+#extraVolumes:
+#  namedVolume:
+#    emptyDir: {}
+#
+#extraVolumeMounts:
+#  namedVolume:
+#    mountPath: /usr/path
+#    readOnly: false
```
Comment (Flydiverny) [inline charts/kubernetes-external-secrets/values.yaml:140]:
```suggestion
extraVolumes: []
#  - name: namedVolume
#    emptyDir: {}
#
extraVolumeMounts: []
# - name: namedVolume
#    mountPath: /usr/path
#    readOnly: false
```

**Review by Flydiverny — COMMENTED** [review]
Think we can keep it even more "standard". At least I think this pattern is pretty common in charts. I could be wrong a bit out of the loop 😄

File: charts/kubernetes-external-secrets/values.yaml (hunk 2 above)
Comment (chrismellard) [reply charts/kubernetes-external-secrets/values.yaml]: All good - more likely me out of the loop ;) Updated.

**Review by Flydiverny — APPROVED** [review]
👍 merging assuming e2e tests doesnt magically fail

**Comment by chrismellard** [comment]
Hold off. spotted one problem

**Comment by chrismellard** [comment]
Resolved. Was outputting volumes yaml in volumeMounts block :)

**Pull request merged**